core/Piwik.php

```diff
@@ -563,7 +563,14 @@ public static function redirectToModule($newModule, $newAction = '', $parameters
      */
     public static function isValidEmailString($emailAddress)
     {
-        return (preg_match('/^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9_.-]+\.[a-zA-Z]{2,}$/D', $emailAddress) > 0);
+        static $zendEmailValidator;
+        if(!$zendEmailValidator) {
+            $zendEmailValidator = new \Zend_Validate_EmailAddress(array(
+                'mx' => false,
+                'deep' => false
+            ));
+        }
```

Comment:
Instead of using a static variable here, DI should be used. You can get an object from DI via,

```
StaticContainer::get('Zend_Validate_EmailAddress'); // full class name goes here
```

If there's no definition in a PHP config file (ie, config/global.php or one of the plugin ones), the DI container will create a new instance itself, autowiring dependencies.

Since you're creating a instance w/ a specific constructor arg, you'll have to add a definition for the instance in `config/global.php`. Should be self-explanatory how to do it, let me know if it isn't.

Regarding `StaticContainer` use: since Piwik's internals are so deeply coupled, we have to use something like this (which stores the container as a global static variable), however we're getting close to removing it. In the future, if possible, try not to use it (by specifying dependencies in the constructor). In this case, since it's a static method, there's no choice but to use it (especially since it's `@api`). If this isn't clear, let me know, I'll try to clarify.